Visual sitemap for Adoptium.net

RounakJoshi09 opened the pull request:
# Description of change
Implemented Issue #434 

I've completed the sitemap implementation for the new adoptium.net website as outlined in the referenced issue. The sitemap now includes all the specified links, along with additional entries for the top ten blog posts and all current authors featured on the site. I've also updated the sitemap.xml generation logic to ensure these links are properly incorporated.
In addition, the sitemap supports search functionality based on issue titles, enhancing discoverability. Throughout the implementation, I’ve followed Adoptium’s established design conventions and color schemes to maintain visual consistency across the site.

## Checklist

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `npm test` and `npm run build` passes
- [x] documentation is changed or added (if applicable)
- [x] permission has been obtained to add new logo (if applicable)
- [x] contribution guidelines followed [here](https://github.com/adoptium/adoptium.net/blob/main/CONTRIBUTING.md)

RounakJoshi09 committed:
feat: Sitemap implemented

Search Fixed

Debouncing and Tests Added

Trnsalation Added for Title,Description in Sitemap

CarmenDelgadoEclipse commented:
Thank you @RounakJoshi09  please review the 3 workflows errors.

gdams reviewed (approved):
Thanks for raising this PR @RounakJoshi09! This is a really well structured set of changes

gdams commented:
@RounakJoshi09 I see one test failure with a snapshot that will need updating. I also see that Copilot has made quite a few suggestions (most of which seem sensible to me). If you wouldn't mind working through those that would be great!

RounakJoshi09 committed:
refactor: Update sitemap functions and improve SitemapClient rendering

- Renamed `getAllSitemapUrls` to `getAllStaticSitemapUrls` for clarity.
- Enhanced `SitemapClient` to use `page.url` as the key in map functions for better uniqueness.
- Simplified URL extraction logic in `generateSitemapData` and added a new `formatBlogTitle` function for consistent blog title formatting.
- Removed unnecessary mock data from tests to streamline the testing process.

RounakJoshi09 committed:
chore: Update sitemap snapshots

- Removed outdated entries from sitemap snapshots to streamline data and ensure accuracy.

xavierfacq committed:
Merge branch 'main' into rounakj/site-map

xavierfacq commented:
Good job! Just I wonder if the section "Authors" is necessary?

RounakJoshi09 commented:
Hi @gdams @xavierfacq , the suggestions from Copilot were quite helpful. I’ve addressed all the flagged conversations by applying the fixes, and I’ve also updated the snapshots to reduce the likelihood of workflow failures. Everything is working as expected locally, and the tests are passing. 
Please take a look when you get a chance and let me know if you have any additional feedback.

Thanks

RounakJoshi09 commented:
Actually @xavierfacq, I initially thought it would be helpful to list all authors/contributers in one place, and the sitemap seemed like a suitable location for that. However, considering the potential growth in the number of blog authors and contributors over time, maintaining this section in the sitemap could become unwieldy . 
If you'd prefer to remove the authors section for now and instead create a dedicated page, or if you have another solution in mind, I'm happy to adjust accordingly and remove it from the sitemap.

Thanks

xavierfacq commented:
I think @CarmenDelgadoEclipse can check if this section is interesting for the website or not. 
Ihmo it's not necessary to index this kind of data.

CarmenDelgadoEclipse commented:
> I think @CarmenDelgadoEclipse can check if this section is interesting for the website or not. Ihmo it's not necessary to index this kind of data.

I agree with @xavierfacq, it's not worth having this section as part of the site map. Maybe it could be a new feature for [the news section](https://adoptium.net/news), [similar to the view we have for the eclipse blog's page](https://blogs.eclipse.org/) sidebar, where you can filter by author. The only thing to consider is that the Adoptium's news section includes both authored blog posts and news shared from other links and repos. And again, not sure it's worth the effort.

RounakJoshi09 committed:
fix: Authors Section Removed

RounakJoshi09 committed:
Merge branch 'main' into rounakj/site-map

RounakJoshi09 committed:
Snaps Updated

RounakJoshi09 commented:
Hi @xavierfacq , Please check my latest commit, on ruuning `npm run update-snapshots`, I got few of these snapshots updated which is not related to my changes but without those changes `npm run coverage` and `npm run test` were not passing on local. Hence I updated them and pushed them. Please check if they are actually needed to be pushed, otherwise I'll remove them.

xavierfacq commented:
> hich is not related to my changes but without those chang

:wave:  Generaly it's related to an update of some packages after a branch update.

If you want to be certain to be up-to-date with the main branch for you merge:

1°/ Update your branch with the main one

2°/ run (or remove node_modules et re-install)
```
# rm -rf node_modules/ && npm install
```

3°/ Update all snapshots by running:
```
# npm run update-snapshots
```

RounakJoshi09 committed:
Unecessary SanpShots Reverted

RounakJoshi09 commented:
Thanks @xavierfacq , that really helped, now we have no extra snapshot changes in this branch, and  `npm run coverage` and `npm run test` are also passing on local 🎉.

xavierfacq commented:
> Thanks @xavierfacq , that really helped, now we have no extra snapshot changes in this branch, and `npm run coverage` and `npm run test` are also passing on local 🎉.

Good job, it's merged !
